lib/datasource/nuget/v3.ts

```diff
@@ -29,97 +39,115 @@ export async function getQueryUrl(url: string): Promise<string | null> {
   }
 
   try {
-    // TODO: fix types
-    const servicesIndexRaw = await http.getJson<any>(url);
-    const searchQueryService = servicesIndexRaw.body.resources.find(
-      (resource) => resource['@type']?.startsWith(resourceType)
-    );
-    const searchQueryServiceId = searchQueryService['@id'];
-
+    const servicesIndexRaw = await http.getJson<ServicesIndexRaw>(url);
+    const services = servicesIndexRaw.body.resources.filter((resource) => {
+      const [type, version] = resource['@type']?.split('/') || [];
+      return type === resourceType && semver.valid(version);
+    });
+    const latestAvailableService = services.pop();
```

Comment:
we should sort services by version, as they can be in any order. 🤔

Reply:
Just wondered if it's safe to assume their ordering 🙃

Reply:
I think we should sort them explicit. Also i think we can create a cache for the whole index, so we don't need a http request per type.

Wonder if we should issue a warning, if we found a version we not yet know, so we don't silently break with future versions.

Reply:
As I can see, we're calling `getResourceUrl` only for `RegistrationsBaseUrl`, so the current version probably is good enough.

Speaking of new version warnings, does it mean we need to hardcode current latest version somewhere around?

Reply:
> As I can see, we're calling `getResourceUrl` only for `RegistrationsBaseUrl`, so the current version probably is good enough.

we need to call it later for `PackageBaseAddress` which is currently hard coded for nuget.org

> Speaking of new version warnings, does it mean we need to hardcode current latest version somewhere around?

yes, maybe somthing like `^3.0.0` is enough for now?